roles/ceph-facts/tasks/facts.yml

```diff
@@ -214,20 +208,42 @@
     - item.value.holders|count == 0
     - item.key is not match osd_auto_discovery_exclude
 
-- name: set_fact rgw_hostname
-  set_fact:
-    rgw_hostname: "{% set _value = ansible_hostname -%}
-    {% for key in (ceph_current_status['servicemap']['services']['rgw']['daemons'] | list) -%}
-    {% if key == ansible_fqdn -%}
-    {% set _value = key -%}
-    {% endif -%}
-    {% endfor -%}
-    {{ _value }}"
+- name: backward compatibility tasks related
   when:
-    - inventory_hostname in groups.get(rgw_group_name, []) or inventory_hostname in groups.get(nfs_group_name, [])
-    - ceph_current_status['servicemap'] is defined
-    - ceph_current_status['servicemap']['services'] is defined
-    - ceph_current_status['servicemap']['services']['rgw'] is defined
+    - groups.get(rgw_group_name, []) | length > 0 or
+      groups.get(nfs_group_name, []) | length > 0
```

Comment:
```suggestion
```

Reply:
in fact you changed the test from` inventory_hostname in groups` to `groups | length > 0`

IMHO we should still use the inventory_hostname condition otherwise the ceph status will be executed on all the nodes when we have rgw nodes present.